src/charm.py

```diff
@@ -184,6 +213,15 @@ def _on_database_changed(self, event: DatabaseCreatedEvent) -> None:
         self.unit.status = MaintenanceStatus("Retrieving database details")
         self._update_container(event)
 
+    def _on_ingress_ready(self, event: IngressPerAppReadyEvent) -> None:
+        if self.unit.is_leader():
+            type = "admin" if event.relation_name == self.admin_ingress.relation_name else "public"
```

Comment:
Also, if handling admin and public ingress is not exactly the same, why not have two different event methods instead of combining them into `_on_ingress_ready`. More code is not a bad thing, what we want to reduce is complexity.